merge_sort_in_c.c

```diff
@@ -0,0 +1,103 @@
+/* C program for Merge Sort */
+#include<stdlib.h> 
+#include<stdio.h> 
+
+// Merges two subarrays of arr[]. 
+// First subarray is arr[l..m] 
+// Second subarray is arr[m+1..r] 
+void merge(int arr[], int l, int m, int r) 
+{ 
+	int i, j, k; 
+	int n1 = m - l + 1; 
+	int n2 = r - m; 
+
+	/* create temp arrays */
+	int L[n1], R[n2]; 
+
+	/* Copy data to temp arrays L[] and R[] */
+	for (i = 0; i < n1; i++) 
+		L[i] = arr[l + i]; 
+	for (j = 0; j < n2; j++) 
+		R[j] = arr[m + 1+ j]; 
+
+	/* Merge the temp arrays back into arr[l..r]*/
+	i = 0; // Initial index of first subarray 
+	j = 0; // Initial index of second subarray 
+	k = l; // Initial index of merged subarray 
+	while (i < n1 && j < n2) 
+	{ 
+		if (L[i] <= R[j]) 
+		{ 
+			arr[k] = L[i]; 
+			i++; 
+		} 
+		else
+		{ 
+			arr[k] = R[j]; 
+			j++; 
+		} 
+		k++; 
+	} 
+
+	/* Copy the remaining elements of L[], if there 
+	are any */
+	while (i < n1) 
+	{ 
+		arr[k] = L[i]; 
+		i++; 
+		k++; 
+	} 
+
+	/* Copy the remaining elements of R[], if there 
+	are any */
+	while (j < n2) 
+	{ 
+		arr[k] = R[j]; 
+		j++; 
+		k++; 
+	} 
+} 
```

Comment:
You can make your code shorter by merging the 3 while loops into 1 using this code:
int i = 0;
    int j = 0; 
    int k = l;
    while(k <= r)
    {
        if(j >= n2 || i < n1 && L[i] <= R[j])
        {
            array[k] = L[i];
            i++;
        }
        else
        {
            array[k] = R[j];
            j++;
        }
        k++;
    }